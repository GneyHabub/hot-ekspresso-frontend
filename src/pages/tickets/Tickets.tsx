import {
  Button, Card, CardContent, CircularProgress, Typography,
} from '@mui/material';
import React from 'react';
import { observer } from 'mobx-react';
import { Box } from '@mui/system';
import AddIcon from '@mui/icons-material/Add';
import { useNavigate } from 'react-router';
import store from '../../store/tickets.store';
import BookingItem from './BookingItem';
import authStore from '../../store/store';

const Tickets: React.FC = observer(() => {
  const navigate = useNavigate();
  const { token } = authStore;

  React.useEffect(() => {
    store.fetchBookings(token);
  }, [token]);

  if (!store.areBookingsFetched) {
    return (
      <Card>
        <CardContent>
          <CircularProgress />
        </CardContent>
      </Card>
    );
  }

  return (
    <>
      <Box
        sx={{
          display: 'flex',
          flexDirection: 'row-reverse',
          paddingBottom: '15px',
        }}
      >
        <Button
          variant="contained"
          startIcon={<AddIcon />}
          onClick={() => navigate('/tickets/new-booking')}
        >
          Create a booking
        </Button>
      </Box>
      <Card>
        <CardContent>
          <Typography variant="h4" gutterBottom component="div">
            My bookings
          </Typography>
          <div>
            {store.bookings.map((b, idx) => (
              <BookingItem idx={idx} booking={b} key={b.id} />
            ))}
          </div>
        </CardContent>
      </Card>
    </>
  );
});

export default Tickets;
