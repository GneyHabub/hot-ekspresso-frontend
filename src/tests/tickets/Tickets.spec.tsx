import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Tickets from '../../pages/tickets/Tickets';

jest.mock('../../store/tickets.store', () => ({
  fetchBookings: jest.fn,
  areBookingsFetched: true,
  bookings: [
    {
      id: 'AAA',
      price: 300,
      flights: [{
        from: 'KZN',
        to: 'GSV',
        departure: new Date('1995-12-17T03:24:00'),
        arrival: new Date('1995-12-17T03:24:00'),
        carrier: 'RedWings',
      }],
      serviceClass: 'Economy',
    },
    {
      id: 'BBB',
      price: 700,
      flights: [
        {
          from: 'GSV',
          to: 'KZN',
          departure: new Date('1995-12-17T03:24:00'),
          arrival: new Date('1995-12-17T03:24:00'),
          carrier: 'RedWings',
        },
        {
          from: 'KZN',
          to: 'DME',
          departure: new Date('1995-12-17T03:24:00'),
          arrival: new Date('1995-12-17T03:24:00'),
          carrier: 'Pobeda',
        },
      ],
      serviceClass: 'Business',
    },
  ],
}));

describe('Login', () => {
  it('matches the snapshot', () => {
    const wrapper = render(<BrowserRouter><Tickets /></BrowserRouter>);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
