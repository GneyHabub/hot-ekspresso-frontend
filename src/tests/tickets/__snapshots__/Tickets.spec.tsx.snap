// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-t66vwn"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Create a booking
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
        >
          My bookings
        </div>
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls="panel1a-content"
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
              id="panel1a-content"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
              >
                <div
                  class="MuiBox-root css-1h2sv0j"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    KZN
                  </p>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="MoreHorizIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    GSV
                  </p>
                </div>
                <span
                  class="MuiTypography-root MuiTypography-overline css-hsgdsv-MuiTypography-root"
                >
                  17.12.1995
                </span>
              </div>
              <div
                class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby="panel1a-content"
                    class="MuiAccordion-region"
                    id="panel1a-content"
                    role="region"
                  >
                    <div
                      class="MuiAccordionDetails-root css-1owrxue-MuiAccordionDetails-root"
                    >
                      <div
                        class="MuiBox-root css-1vgca5b"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                        >
                          Route
                        </h6>
                        <ul
                          class="MuiTimeline-root MuiTimeline-positionRight css-55vwbm-MuiTimeline-root"
                        >
                          <li
                            class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                          >
                            <div
                              class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                            >
                              <span
                                class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                              />
                              <span
                                class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
                              />
                            </div>
                            <div
                              class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-18y2uza-MuiTypography-root-MuiTimelineContent-root"
                            >
                              KZN
                            </div>
                          </li>
                          <li
                            class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                          >
                            <div
                              class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                            >
                              <span
                                class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                              />
                            </div>
                            <div
                              class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1da6d3z-MuiTypography-root-MuiTimelineContent-root"
                            >
                              GSV
                            </div>
                          </li>
                        </ul>
                      </div>
                      <hr
                        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                      />
                      <div
                        class="MuiBox-root css-t6ot1t"
                      >
                        <div
                          class="MuiBox-root css-1v1qs4"
                        >
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Price:
                            </b>
                             
                            â‚¸
                            300
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Departure:
                            </b>
                             
                            17.12.1995 03:24
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Arrival:
                            </b>
                             
                            17.12.1995 03:24
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Service Class:
                            </b>
                             
                            Economy
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls="panel2a-content"
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
              id="panel2a-content"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
              >
                <div
                  class="MuiBox-root css-1h2sv0j"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    GSV
                  </p>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="MoreHorizIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    DME
                  </p>
                </div>
                <span
                  class="MuiTypography-root MuiTypography-overline css-hsgdsv-MuiTypography-root"
                >
                  17.12.1995
                </span>
              </div>
              <div
                class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby="panel2a-content"
                    class="MuiAccordion-region"
                    id="panel2a-content"
                    role="region"
                  >
                    <div
                      class="MuiAccordionDetails-root css-1owrxue-MuiAccordionDetails-root"
                    >
                      <div
                        class="MuiBox-root css-1vgca5b"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                        >
                          Route
                        </h6>
                        <ul
                          class="MuiTimeline-root MuiTimeline-positionRight css-55vwbm-MuiTimeline-root"
                        >
                          <li
                            class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                          >
                            <div
                              class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                            >
                              <span
                                class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                              />
                              <span
                                class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
                              />
                            </div>
                            <div
                              class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-18y2uza-MuiTypography-root-MuiTimelineContent-root"
                            >
                              GSV
                            </div>
                          </li>
                          <li
                            class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                          >
                            <div
                              class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                            >
                              <span
                                class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                              />
                              <span
                                class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
                              />
                            </div>
                            <div
                              class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-18y2uza-MuiTypography-root-MuiTimelineContent-root"
                            >
                              KZN
                            </div>
                          </li>
                          <li
                            class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                          >
                            <div
                              class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                            >
                              <span
                                class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                              />
                            </div>
                            <div
                              class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1da6d3z-MuiTypography-root-MuiTimelineContent-root"
                            >
                              DME
                            </div>
                          </li>
                        </ul>
                      </div>
                      <hr
                        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                      />
                      <div
                        class="MuiBox-root css-t6ot1t"
                      >
                        <div
                          class="MuiBox-root css-1v1qs4"
                        >
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Price:
                            </b>
                             
                            â‚¸
                            700
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Departure:
                            </b>
                             
                            17.12.1995 03:24
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Arrival:
                            </b>
                             
                            17.12.1995 03:24
                          </h6>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                          >
                            <b>
                              Service Class:
                            </b>
                             
                            Business
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
